Don't attempt to format a tuple with "%s"

agashlin opened the pull request:
This fixes #10736, errors in mousemove events over some MSHTML elements
(such as buttons).

<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:

fixes #10736

### Summary of the issue:

mouse move over buttons doesn't read the button text in Internet Explorer 8 MSHTML, fails with a logged TypeError exception.

### Description of how this pull request fixes the issue:

In order to keep the `%` operator from trying to format each element of the Point namedtuple, wrap the position in a one element tuple. This way the right exception gets thrown and caught. I went into more detail in https://github.com/nvaccess/nvda/issues/10736#issuecomment-618760090

### Testing performed:

I confirmed that this allowed the button to be read in the embedded MSHTML instance in the Firefox stub installer. [Build 1 here](https://bugzilla.mozilla.org/show_bug.cgi?id=1596812#c21) can be used to demonstrate the issue, though it has other problems: the Re-install button does not read out when moused over unless this fix is made.

### Known issues with pull request:

There are possibly other similar issues, but I didn't go looking for them.

### Change log entry:

*Bug fixes*
Fix mouse tracking for some MSHTML elements

josephsl commented:
Hi, can you verify that this works in IE11, as IE8 is too old at this time? Thanks.

 

From: Adam Gashlin <notifications@github.com> 
Sent: Thursday, April 23, 2020 7:29 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [nvaccess/nvda] Don't attempt to format a tuple with "%s" (#11050)

 

This fixes #10736 <https://github.com/nvaccess/nvda/issues/10736> , errors in mousemove events over some MSHTML elements
(such as buttons).


Link to issue number:


fixes #10736 <https://github.com/nvaccess/nvda/issues/10736> 


Summary of the issue:


mouse move over buttons doesn't read the button text in Internet Explorer 8 MSHTML, fails with a logged TypeError exception.


Description of how this pull request fixes the issue:


In order to keep the % operator from trying to format each element of the Point namedtuple, wrap the position in a one element tuple. This way the right exception gets thrown and caught. I went into more detail in #10736 (comment) <https://github.com/nvaccess/nvda/issues/10736#issuecomment-618760090> 


Testing performed:


I confirmed that this allowed the button to be read in the embedded MSHTML instance in the Firefox stub installer. Build 1 here <https://bugzilla.mozilla.org/show_bug.cgi?id=1596812#c21>  can be used to demonstrate the issue, though it has other problems: the Re-install button does not read out when moused over unless this fix is made.


Known issues with pull request:


There are possibly other similar issues, but I didn't go looking for them.


Change log entry:


Bug fixes
Fix mouse tracking for some MSHTML elements

  _____  


You can view, comment on, or merge this pull request online at:


  https://github.com/nvaccess/nvda/pull/11050


Commit Summary


*	Don't attempt to format a tuple with "%s"


File Changes


*	M source/NVDAObjects/IAccessible/MSHTML.py <https://github.com/nvaccess/nvda/pull/11050/files#diff-4e46ccec9484e291d412838f24373b76>  (2) 


Patch Links:


*	https://github.com/nvaccess/nvda/pull/11050.patch
*	https://github.com/nvaccess/nvda/pull/11050.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/11050> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXED44GLWXW7DQTPFME3ROD2NHANCNFSM4MPXCPNQ> .

agashlin commented:
> Hi, can you verify that this works in IE11, as IE8 is too old at this time? Thanks.

Yes, I confirmed that this issue is also present in IE11, and this fixes it.

> source/NVDAObjects/IAccessible/MSHTML.py:333:44: E228 missing whitespace around modulo operator

I'll fix this and try running a local flake8.

agashlin committed:
Wrap position in case it was a tuple

This fixes #10736, errors in mousemove events over some MSHTML elements
(such as buttons).

agashlin commented:
I suspect this system test failure is unrelated to my change, as that test uses Chrome and this only touches MSHTML. Is it possible to re-run the system test?

josephsl commented:
Hi, I think a better solution is reverting system tests commit, which will require everyone to rebase their PR’s afterwards. Thanks.

 

From: Adam Gashlin <notifications@github.com> 
Sent: Thursday, April 23, 2020 8:47 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Comment <comment@noreply.github.com>
Subject: Re: [nvaccess/nvda] Don't attempt to format a tuple with "%s" (#11050)

 

I suspect this system test failure is unrelated to my change, as that test uses Chrome and this only touches MSHTML. Is it possible to re-run the system test?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/11050#issuecomment-618785549> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXEAERB5IWWDJ5DJNX6TROEDR3ANCNFSM4MPXCPNQ> .

feerrenrut commented:
> I suspect this system test failure is unrelated to my change

@agashlin Yes we are getting intermittent failures that I haven't identified. I have restarted the build for you.

> I think a better solution is reverting system tests commit

Rather than revert, I think we should just disable until we have time to look at what is causing the failure. I'll look at this today.

feerrenrut reviewed (approved):
Thanks for the investigation and fix @agashlin 

Another approach might be to use an f-string eg: `f"position: {position}"`

feerrenrut commented:
Since this is a small, low risk change, I think it should go into the next release which is currently stalling anyway. I'll change the target to `beta`

feerrenrut commented:
But the PR will need to be rebased onto beta. I'm happy to do that.
